x/params/store/store.go

```diff
@@ -0,0 +1,179 @@
+package store
+
+import (
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Additional capicity to be allocated for Store.space
+// So we don't have to allocate extra space each time appending to the key
+const extraKeyCap = 20
+
+// Individual parameter store for each keeper
+// Transient store persists for a block, so we use it for
+// recording whether the parameter has been changed or not
+type Store struct {
```

Comment:
I don't think we should call this a store... to be consistent I think we should call this a Keeper.  Stores in the SDK are the types/store.go Store, which doesn't require a ctx as first parameter.  To differentiate I think we should call this a Keeper.

Reply:
To differentiate from the global params keeper, we might call that one a params.GlobalKeeper, and it could have .Subkeeper() instead of .Substore().